src/app/plugins/leaderboard.plugin.ts

```diff
@@ -20,18 +20,78 @@ export class LeaderboardPlugin extends Plugin {
   }
 
   public async execute(message: IMessage, args: string[]) {
-    const [gameName, opponent] = args;
+    const [gameName, opponentOne, opponentTwo] = args;
     const gameEnum: Maybe<GameType> = this._convertGameNameToEnum(gameName);
     if (!gameEnum) {
       message.reply(`Couldn't find that game`);
       return;
     }
 
-    if (!opponent) {
+    if (!opponentOne) {
       const embed = await this.container.gameLeaderboardService.createOverallLeaderboardEmbed(
         message.author,
         gameEnum
       );
+
+      message.channel.send(embed);
+      return;
+    } else if (!opponentTwo) {
+      const match = opponentOne.match(/<@!?&?(\d+)>/);
+      if (!match) {
+        message.reply('Invalid <matchup> argument');
+        return;
+      }
+      const uID = match[1];
+      const guild = message.guild;
+      if (!guild) {
+        return;
+      }
+      const oppUser = guild.members.cache.get(uID);
+
+      // user could not be found
+      if (!oppUser) {
+        message.channel.send('User could not be found');
+        return;
+      }
+
+      const embed = await this.container.gameLeaderboardService.createMatchupLeaderboardEmbed(
+        message.author,
+        oppUser.user,
+        gameEnum
+      );
+
+      message.channel.send(embed);
+      return;
+    } else {
```

Comment:
saa
no need to use `else` if we return out of the previous `if`

Reduces indentation, thus complexity